### ISSUE TITLE
InputNumber: Cannot input numbers in AZERTY layout

### ISSUE THREAD

**Issue by justinrlle** [primefaces/primevue#5508]
### Describe the bug

When typing in an `InputNumber` in AZERTY layout, no number can be set, and `undefined` is in fact set in the component.

I'm on MacOS, but not sure it's the issue here, I think the browser abstract away this detail.

### Reproducer

https://primevue.org/inputnumber/

### PrimeVue version

3.50.0

### Vue version

3.x

### Language

TypeScript

### Build / Runtime

Vue CLI App

### Browser(s)

ALL (checked on Chrome and Safari)

### Steps to reproduce the behavior

1. Set your keyboard layout to AZERTY
2. Go to https://primevue.org/inputnumber/
3. Try to type in a number
4. If the input is empty, no number is shown. If there is a number, the value is replaced by `undefined`.

In the video you can see the keys I'm pressing in both `InputNumber` and `InputText`, they're the same and should produce `2020`

https://github.com/primefaces/primevue/assets/10925994/2f14d1e8-e4a0-4fb4-8b9c-8416c3cd1196

### Expected behavior

A number is inputed.

From my quick diving in the code, the issue is with [this lines in the `onInputKeyDown` event](https://github.com/primefaces/primevue/blob/master/components/lib/inputnumber/InputNumber.vue#L379-L383):
```js
            if (event.shiftKey || event.altKey || event.ctrlKey || event.metaKey) {
                this.isSpecialChar = true;

                return;
            }
```
So if a `Shift` key is pressed, the event handler does an early return. Unfortunately, on an AZERTY layout, one must do a combination of `Shift`+_number-key_ to input a number.

**Comment by justinrlle** [comment]
From my tests, it looks like the bug got introduced in `3.50.0`, that is the bug does not exists in `3.49.1`.

Precisely, I have the feeling that the bug appeared with this commit: https://github.com/primefaces/primevue/commit/ec04c03f82f5b602e072f02d695e0582ec0e39a4
At least that's the one which changes the most, and removes the `keypress` event handler which didn't do an early return.

**Comment by kevinbal1** [comment]
I confirm this issue, having it as well, only on AZERTY layouts.

**Comment by melloware** [comment]
This was fixed in PrimeReact: https://github.com/primefaces/primereact/pull/6258

**Comment by arlai-mk** [comment]
> This was fixed in PrimeReact: [primefaces/primereact#6258](https://github.com/primefaces/primereact/pull/6258)

Great, so it seems it should be easy to fix for PrimeVue as well?
Just remove the `event.shiftKey || ` in https://github.com/primefaces/primevue/blob/master/components/lib/inputnumber/InputNumber.vue#L379

If that's the case, are you able to merge the fix for 3.52.0 milestone https://github.com/primefaces/primevue/milestone/178, as I had to rollback to 3.49.1 to get it to work on my app.

Thank you.

**Comment by melloware** [comment]
@arlai-mk yes let me submit a PR
